hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/SqlPlanImpl.java

```diff
@@ -1084,6 +1086,7 @@ static class SelectPlan extends SqlPlanImpl {
             this.isStreaming = isStreaming;
             this.rowMetadata = rowMetadata;
             this.planExecutor = planExecutor;
+            this.ssc = securityContext;
```

Comment:
you cannot cache `securityContext` (`Subject`) in SQL plan cache, it is dependent on the user. We might add it to `PlanKey` but that is a compromise because it would make impossible to reuse plans between users and make the plan cache bigger.